tuf/data/types.go

```diff
@@ -127,6 +128,68 @@ type FileMeta struct {
 	Custom json.RawMessage `json:"custom,omitempty"`
 }
 
+// CheckHashes verifies all the checksums specified by the "hashes" of the payload.
+func CheckHashes(payload []byte, hashes Hashes) error {
+	// k, v indicate the hash algorithm and the corresponding value
+	for k, v := range hashes {
+		switch k {
+		case notary.SHA256:
+			checksum := sha256.Sum256(payload)
+			if subtle.ConstantTimeCompare(checksum[:], v) == 0 {
+				return fmt.Errorf("%s checksum mismatched", k)
+			}
+		case notary.SHA512:
+			checksum := sha512.Sum512(payload)
+			if subtle.ConstantTimeCompare(checksum[:], v) == 0 {
+				return fmt.Errorf("%s checksum mismatched", k)
+			}
+		default:
+			return fmt.Errorf("unsupported hash algorithm")
+		}
+	}
+
+	return nil
+}
+
+// GetSupportedHashes returns the checksums of all the supported hash algorithms
+// of the given payload
+func GetSupportedHashes(payload []byte) Hashes {
+	hashes := make(Hashes)
+
+	for _, v := range notary.NotaryDefaultHashes {
+		switch v {
+		case notary.SHA256:
+			checksum := sha256.Sum256(payload)
+			hashes[v] = checksum[:]
+		case notary.SHA512:
+			checksum := sha512.Sum512(payload)
+			hashes[v] = checksum[:]
+		}
+	}
+
+	return hashes
+}
+
+// SanityCheckForHashes returns an error, or nil, depending on whether
+// the content of the hashes is valid or not. This only check that the
+// checksums are valid in case somebody is drunk.
```

Comment:
Agreed, but if you (@HuKeping) make it out to San Francisco, or a DockerCon, we're totally buying you a beer :-)